astropy/cosmology/funcs.py

```diff
@@ -14,7 +21,41 @@
 __doctest_requires__ = {'*': ['scipy']}
 
 
-def z_at_value(func, fval, zmin=1e-8, zmax=1000, ztol=1e-8, maxfun=500):
+def _z_at_array(func, fvals, zmin, zmax, nbins=10000,
+                logspace=True, interpolation=None):
+    """Helper function to interpolate (func, z) over a grid for array input"""
+    if logspace:
+        zgrid = np.logspace(np.log10(zmin), np.log10(zmax), nbins)
+    else:
+        zgrid = np.linspace(zmin, zmax, nbins)
+
+    fgrid = func(zgrid)
+
+    fvals_val = fvals.value
+    fgrid_val = fgrid.to_value(fvals.unit)
+
+    if interpolation is None:
+        if USE_SCIPY:
+            interpolation = 'cubic'
+        else:
+            warnings.warn("""\
+SciPy not found, so falling back to linear numpy interpolation scheme
+for array z_at_value, instead of the normal cubic spline.""")
```

Comment:
Nitpick: can use strings in parentheses — e.g.("a" "b")— to maintain tab levels. (I think I eyeballed the the line length correctly)

```suggestion
            warnings.warn((
                "SciPy not found, so falling back to linear numpy interpolation"
                " scheme for array z_at_value, instead of the normal cubic"
                " spline."))
```